includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,36 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		$class_prefixes = [
+			'CalendarDay',
+			'CalendarMonth',
+			'CalendarMonthGrid',
+			'DayPicker',
+			'DayPickerKeyboardShortcuts',
+			'DayPickerNavigation',
+			'KeyboardShortcutRow',
```

Comment:
I'm using this function in the console:

```js
function gatherClassNames( elements ) {
	const classNames = new Set();
	const add = ( className ) => {
		if ( !/^i-amphtml-/.test( className ) ) {
			classNames.add( className.replace( /_.+/, '' ) );
		}
	};
	for ( const element of elements ) {
		element.classList.forEach( add );
		gatherClassNames( element.children ).forEach( add );
	}
	return classNames;
}
```

And I'm invoking it via:

```js
gatherClassNames(document.querySelectorAll('amp-date-picker'))
```

I'm not able to find any other class names in use when looking at the `amp-date-picker` examples.

Reply:
Thanks, great function to get the class names.

Reply:
The `KeyboardShortcutRow` class only appears on clicking the `?` icon in the bottom right. But so far, I couldn't find more classes that appear via user interaction.